### ISSUE TITLE
Add support for disabling optional tests by name

### ISSUE THREAD

**Issue by grische** [koalaman/shellcheck#2372]
I would like to enable all optional tests, but one. This does not seem to be currently possible with `check-unassigned-uppercase` as it extends SC2154 and I do not want to disable SC2154 as a whole.

#### For bugs
- Rule Id:  SC2154
- My shellcheck version: 0.8.0 and online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:
```sh
#!/bin/bash

#shellcheck enable=all
#shellcheck disable="check-unassigned-uppercase"

echo "${this_should_fire}"

echo "${THIS_SHOULD_NOT_FIRE}"
```

#### Here's what shellcheck currently says:
```
Line 4:
#shellcheck disable="check-unassigned-uppercase"
                    ^-- SC1125 (error): Invalid key=value pair? Ignoring the rest of this directive starting here.
 
Line 6:
echo "${this_should_fire}"
      ^-- SC2154 (warning): this_should_fire is referenced but not assigned.
 
Line 8:
echo "${THIS_SHOULD_NOT_FIRE}"
      ^-- SC2154 (warning): THIS_SHOULD_NOT_FIRE is referenced but not assigned.
```

#### Here's what I wanted or expected to see:

Only the warning for the uppercase variable, not the lowercase variable.
``` 
Line 8:
echo "${THIS_SHOULD_NOT_FIRE}"
      ^-- SC2154 (warning): THIS_SHOULD_NOT_FIRE is referenced but not assigned.
```
